docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,131 @@
+- Environmental variables can override command line options (except the -o overwrite and -r registry command line options). 
+
+NOTES:
+- Environmental variable overrides remove the need to change the "docker" profile in the res/docker/configuration.toml files - Allowing removal of 50% of the existing configuration.toml files.
+- Environmental variables are considered the record of truth when specified for configuration.  The configuration service (Consul) is the record of truth for all other configuration.
+- The override rules in EdgeX between environmental variables and command line options may be counter intuitive compared to other systems.  There appears to be no standard practice.  Indeed, web searching "Reddit & Starting Fights Env Variables vs Command Line Args" will layout the prevailing differences.
+- Environment variables are named for the configuration element pre-appended with configuration section inclusive of sub-path.  Sub-path "." are replaced with underscores.  Environment variables are in upper camel case.  Here are two examples:
+~~~~~
+Registry_Host  for
+[Registry]
+Host = 'localhost'
+
+Clients_CoreData_Host for
+[Clients]
+  [Clients.CoreData]
+  Host = 'localhost'
+~~~~~
+- Going forward, environmental variables that override command line options should be all uppercase.
+
+All values overriden get logged (indicating which configuration value or op param and the new value).  
+
+## Decision 
+
+These features have been implemented (with some minor changes to be done) for consideration here:  https://github.com/edgexfoundry/go-mod-bootstrap/compare/master...lenny-intel:SelfSeed2.  This code branch will be removed once this ADR is approved and implemented on master.
+
+The implementation for self-seeding services and environmental overrides is already implemented (for Fuji) per this document in the application services and device services (and instituted in the SDKs of each).
+
+## Backward compatibility
+Several aspects of this ADR contain backward compatibility issues for the device service and application service SDKs.  Therefore, for the upcoming minor release, the following guidelines and expections are added to provide for backward compatibility.
+
+- --registry=<url> for Device SDKs
+        
+As earlier versions of the device service SDKs accepted a URI for --registry, if specified on the command line, use the given URI as the address of the configuration provider.  If both --configProvider and --registry specify URIs, then the service should log an error and exit.
+
+- --registry (no ‘=’) and w/o --configProvider for both SDKs
+
+If a configProvider URI isn't specified, but --registry (w/out a URI) is specified, then the service will use the Registry provider information from its local configuration file for both configuration and registry providers.
+
+- Env Var: edgex_registry=<url> for all services (currently has been removed)
```

Comment:
Was this really supported by all services before or just the device SDKs?

Reply:
Just the device services

Reply:
Sorry, no all services used this Env Var.